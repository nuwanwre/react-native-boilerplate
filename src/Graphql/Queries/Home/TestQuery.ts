import gql from 'graphql-tag';

export const testInsertMutation = gql
`
    mutation TestInsertMutation($text: String!) {
        insert_test(objects: {text: $text}) {
            returning {
                id
            }
        }
    }
`;

export const testQuery = gql
`
    query TestQuery($id: uuid!) {
        test(where: {id : {_eq: $id}}) {
            id
            text
        }
    }
`;

export const testDeleteMutation = gql
`
    mutation TestDeleteMutation($id: uuid!) {
        delete_test(where: {id: {_eq: $id}}) {
            returning {
                id
            }
        }
    }
`;
