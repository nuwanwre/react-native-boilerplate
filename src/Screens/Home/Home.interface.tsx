import { WithTranslation } from 'react-i18next';

export namespace IHomePage {
    export interface IProps extends WithTranslation {}

    export interface InitialProps {
        namespacesRequired: string[];
    }

    export interface IStateProps {
        home: {
            version: number;
        };
        image: {
            url: string;
        };
    }
}
