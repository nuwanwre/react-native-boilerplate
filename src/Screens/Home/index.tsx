// #region Global Imports
import * as React from "react";
import { useDispatch } from "react-redux";
import { Image, Dimensions, TouchableOpacity, StyleSheet } from "react-native";
import { useTranslation } from "react-i18next";
// #endregion Global Imports

// #region Local Imports
import RouterActions from "@Services/RouterActions";
import { HomeActions } from "@Actions";
import { Heading } from "@Components";
import {
    Container,
    Apod,
    ApodButton,
    ApodText,
    Middle,
    Centered,
    Buttons,
    LocaleButtonText,
} from "./styled";

import { TestAtom } from '@Components/Atoms';

// #endregion Local Imports

// #region Interface Imports
import { IHomePage } from "@Interfaces";
// #endregion Interface Imports

const pankodLogo = require("static/images/pankod-logo.png");

const styles = StyleSheet.create({
    stretch: {
        width: Dimensions.get("window").width - 50,
        resizeMode: "contain",
    },
});

const Home: React.FunctionComponent<IHomePage.IProps> = () => {
    const { t, i18n } = useTranslation();

    const dispatch = useDispatch();

    const renderLocaleButtons = (activeLanguage: string) =>
        ["en", "es", "tr"].map(lang => (
            <TouchableOpacity
                key={lang}
                onPress={() => i18n.changeLanguage(lang)}>
                <LocaleButtonText isActive={activeLanguage === lang}>
                    {lang}
                </LocaleButtonText>
            </TouchableOpacity>
        ));

    const handleApod = () => {
        dispatch(
            HomeActions.GetApod({
                params: { hd: false },
            })
        );
        RouterActions.push("Apod");
    };

    return (
        <Container>
            {/* <Image style={styles.stretch} source={pankodLogo} /> */}
            <Middle>
                <Centered>
                    {/* <Heading text={t("common:Hello")} /> */}
                    {/* <Buttons>{renderLocaleButtons(i18n.language)}</Buttons> */}
                    <TestAtom />
                </Centered>
                {/* <Apod>
                    <ApodButton onPress={() => handleApod()}>
                        <ApodText>{t("home:Space")}</ApodText>
                    </ApodButton>
                </Apod> */}
            </Middle>
        </Container>
    );
};

export default Home;
