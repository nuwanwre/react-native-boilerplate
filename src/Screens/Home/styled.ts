import styled from 'styled-components/native';

export const Container = styled.View`
    background-color: ${({ theme }) => theme.colors.primary};
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    height: 100%;
`;

export const Middle = styled.View`
    flex: 1;
    justify-content: space-around;
`;

export const Centered = styled.View`
    justify-content: center;
    align-items: center;
`;

export const Buttons = styled.View`
    background: #f9da2e;
    border-radius: 8px;
    letter-spacing: 0;
    width: 100px;
    height: 46px;
    margin-left: 15px;
    padding: 10px 5px;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    color: #4b6c8d;
`;

export const TopText = styled.Text`
    font-size: 20px;
    color: ${({ theme }) => theme.colors.secondary};
    margin-bottom: 10px;
`;
