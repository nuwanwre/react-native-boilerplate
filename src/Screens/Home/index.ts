import Home from './Home.impl';

export default Home;
