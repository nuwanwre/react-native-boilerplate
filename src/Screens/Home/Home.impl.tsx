import React from 'react';
import { useTranslation } from 'react-i18next';

import { TestAtom } from '@Components/Atoms';
import { Container, TopText, Middle, Centered } from './styled';

import { IHomePage } from './Home.interface';

const Home: React.FunctionComponent<IHomePage.IProps> = () => {
    const { t } = useTranslation();

    return (
        <Container key={1}>
            <TopText>{t('home:home')}</TopText>
            <Middle>
                <Centered>
                    <TestAtom />
                </Centered>
            </Middle>
        </Container>
    );
};

export default Home;
