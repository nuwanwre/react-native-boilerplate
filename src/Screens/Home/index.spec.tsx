// #region Global Imports
import * as React from "react";
import { renderWithProviders } from "@Test/Helpers/render";
// #endregion Global Imports

// #region Local Imports
import Home from "./index";
// #endregion Local Imports

describe("Scenes", () => {
    describe("Home", () => {
        it("should match snapshot", () => {
            const wrapper = renderWithProviders(<Home />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
