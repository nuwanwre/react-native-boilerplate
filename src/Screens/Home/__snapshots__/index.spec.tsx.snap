// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scenes Home should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#2c3e50",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "default": Object {},
      }
    }
    style={
      Object {
        "resizeMode": "contain",
        "width": 700,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-around",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "center",
              "position": "relative",
              "textAlign": "center",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 35,
                "fontWeight": "bold",
                "position": "absolute",
              },
            ]
          }
        >
          common:Hello
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "grey",
                "fontSize": 80,
                "fontWeight": "bold",
                "letterSpacing": 1,
                "lineHeight": 86,
                "zIndex": -1,
              },
            ]
          }
        >
          common:Hello
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#f9da2e",
              "borderRadius": 8,
              "color": "#4b6c8d",
              "display": "flex",
              "flexDirection": "row",
              "height": 46,
              "justifyContent": "space-around",
              "letterSpacing": 0,
              "marginLeft": 15,
              "paddingBottom": 10,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 10,
              "width": 100,
            },
          ]
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            isActive={false}
            style={
              Array [
                Object {
                  "color": "#4b6c8d",
                  "fontSize": 20,
                  "fontWeight": "500",
                },
              ]
            }
          >
            en
          </Text>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            isActive={false}
            style={
              Array [
                Object {
                  "color": "#4b6c8d",
                  "fontSize": 20,
                  "fontWeight": "500",
                },
              ]
            }
          >
            es
          </Text>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            isActive={false}
            style={
              Array [
                Object {
                  "color": "#4b6c8d",
                  "fontSize": 20,
                  "fontWeight": "500",
                },
              ]
            }
          >
            tr
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
          },
        ]
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#f9da2e",
            "borderRadius": 5,
            "color": "#2c3e50",
            "opacity": 1,
            "paddingBottom": 5,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 5,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2c3e50",
                "fontSize": 30,
                "fontWeight": "500",
              },
            ]
          }
        >
          home:Space
        </Text>
      </View>
    </View>
  </View>
</View>
`;
